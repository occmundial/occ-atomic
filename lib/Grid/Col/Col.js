import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import useStyles from './styles';

/** The Container component can receive different specifications for any of the breakpoint sizes. It needs to receive an object for the required size, and this object can specify a value for col (Col), offset, push and pull. */
const Col = ({ children, xxs, xs, sm, md, lg, xl, id, className, style }) => {
    const classes = useStyles();

    const getClasses = useCallback((size, data) => {
        const classNames = [
            data.col && classes[`${size}${data.col}`],
            data.offset && classes[`offset-${size}${data.offset}`],
            data.push && classes[`push-${size}${data.push}`],
            data.pull && classes[`pull-${size}${data.pull}`]
        ].filter(Boolean);
        return classnames(...classNames);
    }, [classes]);

    return (
        <div
            className={classnames(
                classes.col,
                xxs && getClasses('xxs', xxs),
                xs && getClasses('xs', xs),
                sm && getClasses('sm', sm),
                md && getClasses('md', md),
                lg && getClasses('lg', lg),
                xl && getClasses('xl', xl),
                className
            )}
            id={id}
            style={style}>
            {children}
        </div>
    );
};

Col.propTypes = {
    children: PropTypes.node,
    /** Specifications for a base window size (from 0px). */
    xxs: PropTypes.object,
    /** Specifications for an extra small window size. */
    xs: PropTypes.object,
    /** Specifications for a small window size. */
    sm: PropTypes.object,
    /** Specifications for a medium window size. */
    md: PropTypes.object,
    /** Specifications for a large window size. */
    lg: PropTypes.object,
    /** Specifications for an extra large window size. */
    xl: PropTypes.object,
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object
};

export default Col;
