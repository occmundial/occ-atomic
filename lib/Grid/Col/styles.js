import grid from '../../subatomic/grid';
import spacing from '../../tokens/spacing.json';

const styles = {
  col: {
    boxSizing: 'border-box',
    position: 'relative',
    minHeight: '1px',
  }
};

const pushGrid = function () {
  const xs = `@media (min-width: ${grid.xs}px)`;
  const sm = `@media (min-width: ${grid.sm}px)`;
  const md = `@media (min-width: ${grid.md}px)`;
  const lg = `@media (min-width: ${grid.lg}px)`;
  const xl = `@media (min-width: ${grid.xl}px)`;
  for (let i = 1; i <= 12; i++) {
    styles[`xxs${i}`] = {
      float: 'left',
      width: `${(100 / 12) * i}%`,
      padding: [0, 6],
    };
    styles[`offset-xxs${i}`] = {
      marginLeft: `${(100 / 12) * i}%`
    };
    styles[`push-xxs${i}`] = {
      left: `${(100 / 12) * i}%`
    };
    styles[`pull-xxs${i}`] = {
      right: `${(100 / 12) * i}%`
    };
  }
  styles[xs] = {};
  styles[sm] = {};
  styles[md] = {};
  styles[lg] = {};
  styles[xl] = {};
  for (let i = 1; i <= 12; i++) {
    styles[xs][`xs${i}`] = {
      float: 'left',
      width: `${(100 / 12) * i}%`,
      padding: [0, spacing['size-2']],
    };
    styles[xs][`offset-xs${i}`] = {
      marginLeft: `${(100 / 12) * i}%`
    };
    styles[xs][`push-xs${i}`] = {
      left: `${(100 / 12) * i}%`
    };
    styles[xs][`pull-xs${i}`] = {
      right: `${(100 / 12) * i}%`
    };
    styles[sm][`sm${i}`] = {
      float: 'left',
      width: `${(100 / 12) * i}%`,
      padding: [0, spacing['size-3']],
    };
    styles[sm][`offset-sm${i}`] = {
      marginLeft: `${(100 / 12) * i}%`
    };
    styles[sm][`push-sm${i}`] = {
      left: `${(100 / 12) * i}%`
    };
    styles[sm][`pull-sm${i}`] = {
      right: `${(100 / 12) * i}%`
    };
    styles[md][`md${i}`] = {
      float: 'left',
      width: `${(100 / 12) * i}%`
    };
    styles[md][`offset-md${i}`] = {
      marginLeft: `${(100 / 12) * i}%`
    };
    styles[md][`push-md${i}`] = {
      left: `${(100 / 12) * i}%`
    };
    styles[md][`pull-md${i}`] = {
      right: `${(100 / 12) * i}%`
    };
    styles[lg][`lg${i}`] = {
      float: 'left',
      width: `${(100 / 12) * i}%`
    };
    styles[lg][`offset-lg${i}`] = {
      marginLeft: `${(100 / 12) * i}%`
    };
    styles[lg][`push-lg${i}`] = {
      left: `${(100 / 12) * i}%`
    };
    styles[lg][`pull-lg${i}`] = {
      right: `${(100 / 12) * i}%`
    };
    styles[xl][`xl${i}`] = {
      float: 'left',
      width: `${(100 / 12) * i}%`
    };
    styles[xl][`offset-xl${i}`] = {
      marginLeft: `${(100 / 12) * i}%`
    };
    styles[xl][`push-xl${i}`] = {
      left: `${(100 / 12) * i}%`
    };
    styles[xl][`pull-xl${i}`] = {
      right: `${(100 / 12) * i}%`
    };
  }
};

pushGrid();

export default styles;
