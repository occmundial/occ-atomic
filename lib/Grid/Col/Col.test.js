import React from 'react';
import { mount } from 'enzyme';
import { JssProvider, SheetsRegistry } from 'react-jss';
import classnames from 'classnames';

import Col from './Col';

describe("Col", () => {

    it('matches the snapshot', () => {
        const wrapper = mount(
            <Col xs={{ col: 12 }} />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches the classes', () => {
        const sheets = new SheetsRegistry();
        const wrapper = mount(
            <JssProvider registry={sheets}>
                <Col xxs={{ col: 6, offset: 3, push: 1, pull: 1 }} />
            </JssProvider>
        );
        const { classes } = sheets.registry[0];
        expect(wrapper.find('div').prop('className')).toEqual(classnames(
            classes.col,
            classes.xxs6,
            classes['offset-xxs3'],
            classes['push-xxs1'],
            classes['pull-xxs1']
        ));
    });

});
