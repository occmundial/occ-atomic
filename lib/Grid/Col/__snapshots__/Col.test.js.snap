// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`Col styles matches the snapshot 1`] = `
Object {
  "@media (min-width: 1200px)": Object {
    "lg1": Object {
      "float": "left",
      "width": "8.333333333333334%",
    },
    "lg10": Object {
      "float": "left",
      "width": "83.33333333333334%",
    },
    "lg11": Object {
      "float": "left",
      "width": "91.66666666666667%",
    },
    "lg12": Object {
      "float": "left",
      "width": "100%",
    },
    "lg2": Object {
      "float": "left",
      "width": "16.666666666666668%",
    },
    "lg3": Object {
      "float": "left",
      "width": "25%",
    },
    "lg4": Object {
      "float": "left",
      "width": "33.333333333333336%",
    },
    "lg5": Object {
      "float": "left",
      "width": "41.66666666666667%",
    },
    "lg6": Object {
      "float": "left",
      "width": "50%",
    },
    "lg7": Object {
      "float": "left",
      "width": "58.333333333333336%",
    },
    "lg8": Object {
      "float": "left",
      "width": "66.66666666666667%",
    },
    "lg9": Object {
      "float": "left",
      "width": "75%",
    },
    "offset-lg1": Object {
      "marginLeft": "8.333333333333334%",
    },
    "offset-lg10": Object {
      "marginLeft": "83.33333333333334%",
    },
    "offset-lg11": Object {
      "marginLeft": "91.66666666666667%",
    },
    "offset-lg12": Object {
      "marginLeft": "100%",
    },
    "offset-lg2": Object {
      "marginLeft": "16.666666666666668%",
    },
    "offset-lg3": Object {
      "marginLeft": "25%",
    },
    "offset-lg4": Object {
      "marginLeft": "33.333333333333336%",
    },
    "offset-lg5": Object {
      "marginLeft": "41.66666666666667%",
    },
    "offset-lg6": Object {
      "marginLeft": "50%",
    },
    "offset-lg7": Object {
      "marginLeft": "58.333333333333336%",
    },
    "offset-lg8": Object {
      "marginLeft": "66.66666666666667%",
    },
    "offset-lg9": Object {
      "marginLeft": "75%",
    },
    "pull-lg1": Object {
      "right": "8.333333333333334%",
    },
    "pull-lg10": Object {
      "right": "83.33333333333334%",
    },
    "pull-lg11": Object {
      "right": "91.66666666666667%",
    },
    "pull-lg12": Object {
      "right": "100%",
    },
    "pull-lg2": Object {
      "right": "16.666666666666668%",
    },
    "pull-lg3": Object {
      "right": "25%",
    },
    "pull-lg4": Object {
      "right": "33.333333333333336%",
    },
    "pull-lg5": Object {
      "right": "41.66666666666667%",
    },
    "pull-lg6": Object {
      "right": "50%",
    },
    "pull-lg7": Object {
      "right": "58.333333333333336%",
    },
    "pull-lg8": Object {
      "right": "66.66666666666667%",
    },
    "pull-lg9": Object {
      "right": "75%",
    },
    "push-lg1": Object {
      "left": "8.333333333333334%",
    },
    "push-lg10": Object {
      "left": "83.33333333333334%",
    },
    "push-lg11": Object {
      "left": "91.66666666666667%",
    },
    "push-lg12": Object {
      "left": "100%",
    },
    "push-lg2": Object {
      "left": "16.666666666666668%",
    },
    "push-lg3": Object {
      "left": "25%",
    },
    "push-lg4": Object {
      "left": "33.333333333333336%",
    },
    "push-lg5": Object {
      "left": "41.66666666666667%",
    },
    "push-lg6": Object {
      "left": "50%",
    },
    "push-lg7": Object {
      "left": "58.333333333333336%",
    },
    "push-lg8": Object {
      "left": "66.66666666666667%",
    },
    "push-lg9": Object {
      "left": "75%",
    },
  },
  "@media (min-width: 1500px)": Object {
    "offset-xl1": Object {
      "marginLeft": "8.333333333333334%",
    },
    "offset-xl10": Object {
      "marginLeft": "83.33333333333334%",
    },
    "offset-xl11": Object {
      "marginLeft": "91.66666666666667%",
    },
    "offset-xl12": Object {
      "marginLeft": "100%",
    },
    "offset-xl2": Object {
      "marginLeft": "16.666666666666668%",
    },
    "offset-xl3": Object {
      "marginLeft": "25%",
    },
    "offset-xl4": Object {
      "marginLeft": "33.333333333333336%",
    },
    "offset-xl5": Object {
      "marginLeft": "41.66666666666667%",
    },
    "offset-xl6": Object {
      "marginLeft": "50%",
    },
    "offset-xl7": Object {
      "marginLeft": "58.333333333333336%",
    },
    "offset-xl8": Object {
      "marginLeft": "66.66666666666667%",
    },
    "offset-xl9": Object {
      "marginLeft": "75%",
    },
    "pull-xl1": Object {
      "right": "8.333333333333334%",
    },
    "pull-xl10": Object {
      "right": "83.33333333333334%",
    },
    "pull-xl11": Object {
      "right": "91.66666666666667%",
    },
    "pull-xl12": Object {
      "right": "100%",
    },
    "pull-xl2": Object {
      "right": "16.666666666666668%",
    },
    "pull-xl3": Object {
      "right": "25%",
    },
    "pull-xl4": Object {
      "right": "33.333333333333336%",
    },
    "pull-xl5": Object {
      "right": "41.66666666666667%",
    },
    "pull-xl6": Object {
      "right": "50%",
    },
    "pull-xl7": Object {
      "right": "58.333333333333336%",
    },
    "pull-xl8": Object {
      "right": "66.66666666666667%",
    },
    "pull-xl9": Object {
      "right": "75%",
    },
    "push-xl1": Object {
      "left": "8.333333333333334%",
    },
    "push-xl10": Object {
      "left": "83.33333333333334%",
    },
    "push-xl11": Object {
      "left": "91.66666666666667%",
    },
    "push-xl12": Object {
      "left": "100%",
    },
    "push-xl2": Object {
      "left": "16.666666666666668%",
    },
    "push-xl3": Object {
      "left": "25%",
    },
    "push-xl4": Object {
      "left": "33.333333333333336%",
    },
    "push-xl5": Object {
      "left": "41.66666666666667%",
    },
    "push-xl6": Object {
      "left": "50%",
    },
    "push-xl7": Object {
      "left": "58.333333333333336%",
    },
    "push-xl8": Object {
      "left": "66.66666666666667%",
    },
    "push-xl9": Object {
      "left": "75%",
    },
    "xl1": Object {
      "float": "left",
      "width": "8.333333333333334%",
    },
    "xl10": Object {
      "float": "left",
      "width": "83.33333333333334%",
    },
    "xl11": Object {
      "float": "left",
      "width": "91.66666666666667%",
    },
    "xl12": Object {
      "float": "left",
      "width": "100%",
    },
    "xl2": Object {
      "float": "left",
      "width": "16.666666666666668%",
    },
    "xl3": Object {
      "float": "left",
      "width": "25%",
    },
    "xl4": Object {
      "float": "left",
      "width": "33.333333333333336%",
    },
    "xl5": Object {
      "float": "left",
      "width": "41.66666666666667%",
    },
    "xl6": Object {
      "float": "left",
      "width": "50%",
    },
    "xl7": Object {
      "float": "left",
      "width": "58.333333333333336%",
    },
    "xl8": Object {
      "float": "left",
      "width": "66.66666666666667%",
    },
    "xl9": Object {
      "float": "left",
      "width": "75%",
    },
  },
  "@media (min-width: 576px)": Object {
    "offset-xs1": Object {
      "marginLeft": "8.333333333333334%",
    },
    "offset-xs10": Object {
      "marginLeft": "83.33333333333334%",
    },
    "offset-xs11": Object {
      "marginLeft": "91.66666666666667%",
    },
    "offset-xs12": Object {
      "marginLeft": "100%",
    },
    "offset-xs2": Object {
      "marginLeft": "16.666666666666668%",
    },
    "offset-xs3": Object {
      "marginLeft": "25%",
    },
    "offset-xs4": Object {
      "marginLeft": "33.333333333333336%",
    },
    "offset-xs5": Object {
      "marginLeft": "41.66666666666667%",
    },
    "offset-xs6": Object {
      "marginLeft": "50%",
    },
    "offset-xs7": Object {
      "marginLeft": "58.333333333333336%",
    },
    "offset-xs8": Object {
      "marginLeft": "66.66666666666667%",
    },
    "offset-xs9": Object {
      "marginLeft": "75%",
    },
    "pull-xs1": Object {
      "right": "8.333333333333334%",
    },
    "pull-xs10": Object {
      "right": "83.33333333333334%",
    },
    "pull-xs11": Object {
      "right": "91.66666666666667%",
    },
    "pull-xs12": Object {
      "right": "100%",
    },
    "pull-xs2": Object {
      "right": "16.666666666666668%",
    },
    "pull-xs3": Object {
      "right": "25%",
    },
    "pull-xs4": Object {
      "right": "33.333333333333336%",
    },
    "pull-xs5": Object {
      "right": "41.66666666666667%",
    },
    "pull-xs6": Object {
      "right": "50%",
    },
    "pull-xs7": Object {
      "right": "58.333333333333336%",
    },
    "pull-xs8": Object {
      "right": "66.66666666666667%",
    },
    "pull-xs9": Object {
      "right": "75%",
    },
    "push-xs1": Object {
      "left": "8.333333333333334%",
    },
    "push-xs10": Object {
      "left": "83.33333333333334%",
    },
    "push-xs11": Object {
      "left": "91.66666666666667%",
    },
    "push-xs12": Object {
      "left": "100%",
    },
    "push-xs2": Object {
      "left": "16.666666666666668%",
    },
    "push-xs3": Object {
      "left": "25%",
    },
    "push-xs4": Object {
      "left": "33.333333333333336%",
    },
    "push-xs5": Object {
      "left": "41.66666666666667%",
    },
    "push-xs6": Object {
      "left": "50%",
    },
    "push-xs7": Object {
      "left": "58.333333333333336%",
    },
    "push-xs8": Object {
      "left": "66.66666666666667%",
    },
    "push-xs9": Object {
      "left": "75%",
    },
    "xs1": Object {
      "float": "left",
      "width": "8.333333333333334%",
    },
    "xs10": Object {
      "float": "left",
      "width": "83.33333333333334%",
    },
    "xs11": Object {
      "float": "left",
      "width": "91.66666666666667%",
    },
    "xs12": Object {
      "float": "left",
      "width": "100%",
    },
    "xs2": Object {
      "float": "left",
      "width": "16.666666666666668%",
    },
    "xs3": Object {
      "float": "left",
      "width": "25%",
    },
    "xs4": Object {
      "float": "left",
      "width": "33.333333333333336%",
    },
    "xs5": Object {
      "float": "left",
      "width": "41.66666666666667%",
    },
    "xs6": Object {
      "float": "left",
      "width": "50%",
    },
    "xs7": Object {
      "float": "left",
      "width": "58.333333333333336%",
    },
    "xs8": Object {
      "float": "left",
      "width": "66.66666666666667%",
    },
    "xs9": Object {
      "float": "left",
      "width": "75%",
    },
  },
  "@media (min-width: 768px)": Object {
    "offset-sm1": Object {
      "marginLeft": "8.333333333333334%",
    },
    "offset-sm10": Object {
      "marginLeft": "83.33333333333334%",
    },
    "offset-sm11": Object {
      "marginLeft": "91.66666666666667%",
    },
    "offset-sm12": Object {
      "marginLeft": "100%",
    },
    "offset-sm2": Object {
      "marginLeft": "16.666666666666668%",
    },
    "offset-sm3": Object {
      "marginLeft": "25%",
    },
    "offset-sm4": Object {
      "marginLeft": "33.333333333333336%",
    },
    "offset-sm5": Object {
      "marginLeft": "41.66666666666667%",
    },
    "offset-sm6": Object {
      "marginLeft": "50%",
    },
    "offset-sm7": Object {
      "marginLeft": "58.333333333333336%",
    },
    "offset-sm8": Object {
      "marginLeft": "66.66666666666667%",
    },
    "offset-sm9": Object {
      "marginLeft": "75%",
    },
    "pull-sm1": Object {
      "right": "8.333333333333334%",
    },
    "pull-sm10": Object {
      "right": "83.33333333333334%",
    },
    "pull-sm11": Object {
      "right": "91.66666666666667%",
    },
    "pull-sm12": Object {
      "right": "100%",
    },
    "pull-sm2": Object {
      "right": "16.666666666666668%",
    },
    "pull-sm3": Object {
      "right": "25%",
    },
    "pull-sm4": Object {
      "right": "33.333333333333336%",
    },
    "pull-sm5": Object {
      "right": "41.66666666666667%",
    },
    "pull-sm6": Object {
      "right": "50%",
    },
    "pull-sm7": Object {
      "right": "58.333333333333336%",
    },
    "pull-sm8": Object {
      "right": "66.66666666666667%",
    },
    "pull-sm9": Object {
      "right": "75%",
    },
    "push-sm1": Object {
      "left": "8.333333333333334%",
    },
    "push-sm10": Object {
      "left": "83.33333333333334%",
    },
    "push-sm11": Object {
      "left": "91.66666666666667%",
    },
    "push-sm12": Object {
      "left": "100%",
    },
    "push-sm2": Object {
      "left": "16.666666666666668%",
    },
    "push-sm3": Object {
      "left": "25%",
    },
    "push-sm4": Object {
      "left": "33.333333333333336%",
    },
    "push-sm5": Object {
      "left": "41.66666666666667%",
    },
    "push-sm6": Object {
      "left": "50%",
    },
    "push-sm7": Object {
      "left": "58.333333333333336%",
    },
    "push-sm8": Object {
      "left": "66.66666666666667%",
    },
    "push-sm9": Object {
      "left": "75%",
    },
    "sm1": Object {
      "float": "left",
      "width": "8.333333333333334%",
    },
    "sm10": Object {
      "float": "left",
      "width": "83.33333333333334%",
    },
    "sm11": Object {
      "float": "left",
      "width": "91.66666666666667%",
    },
    "sm12": Object {
      "float": "left",
      "width": "100%",
    },
    "sm2": Object {
      "float": "left",
      "width": "16.666666666666668%",
    },
    "sm3": Object {
      "float": "left",
      "width": "25%",
    },
    "sm4": Object {
      "float": "left",
      "width": "33.333333333333336%",
    },
    "sm5": Object {
      "float": "left",
      "width": "41.66666666666667%",
    },
    "sm6": Object {
      "float": "left",
      "width": "50%",
    },
    "sm7": Object {
      "float": "left",
      "width": "58.333333333333336%",
    },
    "sm8": Object {
      "float": "left",
      "width": "66.66666666666667%",
    },
    "sm9": Object {
      "float": "left",
      "width": "75%",
    },
  },
  "@media (min-width: 992px)": Object {
    "md1": Object {
      "float": "left",
      "width": "8.333333333333334%",
    },
    "md10": Object {
      "float": "left",
      "width": "83.33333333333334%",
    },
    "md11": Object {
      "float": "left",
      "width": "91.66666666666667%",
    },
    "md12": Object {
      "float": "left",
      "width": "100%",
    },
    "md2": Object {
      "float": "left",
      "width": "16.666666666666668%",
    },
    "md3": Object {
      "float": "left",
      "width": "25%",
    },
    "md4": Object {
      "float": "left",
      "width": "33.333333333333336%",
    },
    "md5": Object {
      "float": "left",
      "width": "41.66666666666667%",
    },
    "md6": Object {
      "float": "left",
      "width": "50%",
    },
    "md7": Object {
      "float": "left",
      "width": "58.333333333333336%",
    },
    "md8": Object {
      "float": "left",
      "width": "66.66666666666667%",
    },
    "md9": Object {
      "float": "left",
      "width": "75%",
    },
    "offset-md1": Object {
      "marginLeft": "8.333333333333334%",
    },
    "offset-md10": Object {
      "marginLeft": "83.33333333333334%",
    },
    "offset-md11": Object {
      "marginLeft": "91.66666666666667%",
    },
    "offset-md12": Object {
      "marginLeft": "100%",
    },
    "offset-md2": Object {
      "marginLeft": "16.666666666666668%",
    },
    "offset-md3": Object {
      "marginLeft": "25%",
    },
    "offset-md4": Object {
      "marginLeft": "33.333333333333336%",
    },
    "offset-md5": Object {
      "marginLeft": "41.66666666666667%",
    },
    "offset-md6": Object {
      "marginLeft": "50%",
    },
    "offset-md7": Object {
      "marginLeft": "58.333333333333336%",
    },
    "offset-md8": Object {
      "marginLeft": "66.66666666666667%",
    },
    "offset-md9": Object {
      "marginLeft": "75%",
    },
    "pull-md1": Object {
      "right": "8.333333333333334%",
    },
    "pull-md10": Object {
      "right": "83.33333333333334%",
    },
    "pull-md11": Object {
      "right": "91.66666666666667%",
    },
    "pull-md12": Object {
      "right": "100%",
    },
    "pull-md2": Object {
      "right": "16.666666666666668%",
    },
    "pull-md3": Object {
      "right": "25%",
    },
    "pull-md4": Object {
      "right": "33.333333333333336%",
    },
    "pull-md5": Object {
      "right": "41.66666666666667%",
    },
    "pull-md6": Object {
      "right": "50%",
    },
    "pull-md7": Object {
      "right": "58.333333333333336%",
    },
    "pull-md8": Object {
      "right": "66.66666666666667%",
    },
    "pull-md9": Object {
      "right": "75%",
    },
    "push-md1": Object {
      "left": "8.333333333333334%",
    },
    "push-md10": Object {
      "left": "83.33333333333334%",
    },
    "push-md11": Object {
      "left": "91.66666666666667%",
    },
    "push-md12": Object {
      "left": "100%",
    },
    "push-md2": Object {
      "left": "16.666666666666668%",
    },
    "push-md3": Object {
      "left": "25%",
    },
    "push-md4": Object {
      "left": "33.333333333333336%",
    },
    "push-md5": Object {
      "left": "41.66666666666667%",
    },
    "push-md6": Object {
      "left": "50%",
    },
    "push-md7": Object {
      "left": "58.333333333333336%",
    },
    "push-md8": Object {
      "left": "66.66666666666667%",
    },
    "push-md9": Object {
      "left": "75%",
    },
  },
  "col": Object {
    "boxSizing": "border-box",
    "minHeight": "1px",
    "paddingLeft": 12,
    "paddingRight": 12,
    "position": "relative",
  },
  "offset-xxs1": Object {
    "marginLeft": "8.333333333333334%",
  },
  "offset-xxs10": Object {
    "marginLeft": "83.33333333333334%",
  },
  "offset-xxs11": Object {
    "marginLeft": "91.66666666666667%",
  },
  "offset-xxs12": Object {
    "marginLeft": "100%",
  },
  "offset-xxs2": Object {
    "marginLeft": "16.666666666666668%",
  },
  "offset-xxs3": Object {
    "marginLeft": "25%",
  },
  "offset-xxs4": Object {
    "marginLeft": "33.333333333333336%",
  },
  "offset-xxs5": Object {
    "marginLeft": "41.66666666666667%",
  },
  "offset-xxs6": Object {
    "marginLeft": "50%",
  },
  "offset-xxs7": Object {
    "marginLeft": "58.333333333333336%",
  },
  "offset-xxs8": Object {
    "marginLeft": "66.66666666666667%",
  },
  "offset-xxs9": Object {
    "marginLeft": "75%",
  },
  "pull-xxs1": Object {
    "right": "8.333333333333334%",
  },
  "pull-xxs10": Object {
    "right": "83.33333333333334%",
  },
  "pull-xxs11": Object {
    "right": "91.66666666666667%",
  },
  "pull-xxs12": Object {
    "right": "100%",
  },
  "pull-xxs2": Object {
    "right": "16.666666666666668%",
  },
  "pull-xxs3": Object {
    "right": "25%",
  },
  "pull-xxs4": Object {
    "right": "33.333333333333336%",
  },
  "pull-xxs5": Object {
    "right": "41.66666666666667%",
  },
  "pull-xxs6": Object {
    "right": "50%",
  },
  "pull-xxs7": Object {
    "right": "58.333333333333336%",
  },
  "pull-xxs8": Object {
    "right": "66.66666666666667%",
  },
  "pull-xxs9": Object {
    "right": "75%",
  },
  "push-xxs1": Object {
    "left": "8.333333333333334%",
  },
  "push-xxs10": Object {
    "left": "83.33333333333334%",
  },
  "push-xxs11": Object {
    "left": "91.66666666666667%",
  },
  "push-xxs12": Object {
    "left": "100%",
  },
  "push-xxs2": Object {
    "left": "16.666666666666668%",
  },
  "push-xxs3": Object {
    "left": "25%",
  },
  "push-xxs4": Object {
    "left": "33.333333333333336%",
  },
  "push-xxs5": Object {
    "left": "41.66666666666667%",
  },
  "push-xxs6": Object {
    "left": "50%",
  },
  "push-xxs7": Object {
    "left": "58.333333333333336%",
  },
  "push-xxs8": Object {
    "left": "66.66666666666667%",
  },
  "push-xxs9": Object {
    "left": "75%",
  },
  "xxs1": Object {
    "float": "left",
    "width": "8.333333333333334%",
  },
  "xxs10": Object {
    "float": "left",
    "width": "83.33333333333334%",
  },
  "xxs11": Object {
    "float": "left",
    "width": "91.66666666666667%",
  },
  "xxs12": Object {
    "float": "left",
    "width": "100%",
  },
  "xxs2": Object {
    "float": "left",
    "width": "16.666666666666668%",
  },
  "xxs3": Object {
    "float": "left",
    "width": "25%",
  },
  "xxs4": Object {
    "float": "left",
    "width": "33.333333333333336%",
  },
  "xxs5": Object {
    "float": "left",
    "width": "41.66666666666667%",
  },
  "xxs6": Object {
    "float": "left",
    "width": "50%",
  },
  "xxs7": Object {
    "float": "left",
    "width": "58.333333333333336%",
  },
  "xxs8": Object {
    "float": "left",
    "width": "66.66666666666667%",
  },
  "xxs9": Object {
    "float": "left",
    "width": "75%",
  },
}
`;

exports[`ColJSS matches the snapshot 1`] = `ShallowWrapper {}`;
